File: 217/main.py
class Solution(object):
    def containsDuplicate(self, nums):
        seen = [] #empty set to store seen x numbers
        for x in nums:
            if x in seen:
                return True
            seen.append(x)
        return False


if __name__ == "__main__":
    sol = Solution()
    print(sol.containsDuplicate([1, 2, 3, 1]))  #Expected: True
    print(sol.containsDuplicate([1, 2, 3, 4]))  #Expected: False

File: 217/main.cpp
#include "bits/stdc++.h"
using namespace std;

class Solution {
public:
    void containsDuplicate(vector<int>& nums) {
        //initialize empty hash set
        unordered_set<int> us;
        //search through nums vector
        for(auto x : nums){
            auto it = us.find(x); // O(1)
            if (it != us.end())
                cout << "true" << endl;
            else 
                us.insert(x);          
        }
    }
};

int main() {
    Solution sol;
    int n;
    cin>>n;
    vector<int> nums (n);
    for (int i = 0; i < n; i++){
        cin >> nums[i];
    }
    sol.containsDuplicate(nums);
    return 0;
}